stats/src/test/java/io/airlift/stats/cardinality/TestSparseHll.java

```diff
@@ -57,6 +59,24 @@ public void testToDense(int prefixBitLength)
         verifyToDense(prefixBitLength, ImmutableList.of(224L, 271L));
     }
 
+    @Test
+    public void testRetainedSize()
+    {
+        SparseHll sparseHll = new SparseHll(10);
+        // use an empty array to mock the entries in SparseHll
+        int[] entries = new int[1];
+        long retainedSize = sizeOf(entries) + ClassLayout.parseClass(SparseHll.class).instanceSize();
+        for (int value = 0; value < 100; value++) {
+            sparseHll.insertHash(Murmur3Hash128.hash64(value));
+            if (value % 10 == 1) {
+                // we increase the capacity by 10 once full
+                entries = new int[value + 10];
+                retainedSize = sizeOf(entries) + ClassLayout.parseClass(SparseHll.class).instanceSize();
```

Comment:
Computing ClassLayout.parseClass(SparseHll.class).instanceSize() can be expensive. Move it out of the loop (ideally, to a constant).